src/main/kotlin/me/partlysanestudios/partlysaneskies/features/farming/garden/VisitorTradeValue.kt

```diff
@@ -0,0 +1,267 @@
+            if (costStartIndex == -1) {
+//                If there is something there, it's an item with only one
+                if (costLine.length > 5) {
+                    singleItem = true
+                    costStartIndex = costLine.length
+                } else {
+                    continue
+                }
+            }
+
+            // Gets the name of ihe item and formats it
+            var name = costLine.substring(0, costStartIndex)
+            name = stripLeading(name)
+            name = stripTrailing(name)
+            var amount: Int
+            if (singleItem) {
+                amount = 1
+            } else {
+                // Gets the cost of the item and converts it to an integer
+                amountString = costLine.substring(costStartIndex + 1)
+                // Replaces all non-numeric characters in the string
+                amountString = amountString.replace("[^\\d.]".toRegex(), "")
+                amountString = amountString.replace(",", "")
+                amountString = amountString.replace(".", "")
+                amount = amountString.toInt()
+            }
+
+            // Adds it to the cost map
+            costMap[name] = amount
+        }
+        return costMap
+    }
+
+    private fun getRewardsLore(): List<String> {
+        val inventories = minecraft.currentScreen.getSeparateUpperLowerInventories()
+        val trader = inventories[0] ?: return ArrayList<String>()
+
+        // Slots 29 is where the accept buttons is
+        val acceptButton = trader.getStackInSlot(29)
+        val formattedAcceptButtonLore = acceptButton.getLore()
+
+        // Removes all the format codes from lore
+        val unformattedAcceptButtonLore = removeColorCodesFromList(formattedAcceptButtonLore)
+        val rewardsStartIndex = unformattedAcceptButtonLore.indexOf("Rewards:")
+        return formattedAcceptButtonLore.subList(rewardsStartIndex, formattedAcceptButtonLore.size)
+    }
+
+    private fun getCopperReturn(): Int {
+        val unformattedRewardsLore: List<String> = removeColorCodesFromList(getRewardsLore())
+        for (line in unformattedRewardsLore) {
+            if (!line.contains(" Copper")) {
+                continue
+            }
+            var strippedLine = stripLeading(line)
+            strippedLine = stripTrailing(strippedLine)
+            val amountStartIndex = strippedLine.indexOf("+") + 1
+            val amountEndIndex = strippedLine.indexOf(" C")
+            val amountString = strippedLine.substring(amountStartIndex, amountEndIndex)
+            return amountString.toInt()
+        }
+        return -1
+    }
+
+    private fun getItemCost(itemId: String, quantity: Int): Double {
+        return quantity * (getItem(itemId)?.getBuyPrice() ?: return 0.0)
+    }
+
+    // Returns a new list with all format codes removed
+    private fun removeColorCodesFromList(list: List<String>): ArrayList<String> {
```

Comment:
Make this a List<String>.removeColorCodes() pls 
You also don't need to specify the word list in the name then